File: src/main/java/com/skyspaceoneoneone/service/AircraftService.java
package com.skyspaceoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceoneoneone.domain.Aircraft;
import com.skyspaceoneoneone.dto.AircraftDTO;
import com.skyspaceoneoneone.dto.AircraftSearchDTO;
import com.skyspaceoneoneone.dto.AircraftPageDTO;
import com.skyspaceoneoneone.dto.AircraftConvertCriteriaDTO;
import com.skyspaceoneoneone.service.GenericService;
import com.skyspaceoneoneone.dto.common.RequestDTO;
import com.skyspaceoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AircraftService extends GenericService<Aircraft, Integer> {

	List<Aircraft> findAll();

	ResultDTO addAircraft(AircraftDTO aircraftDTO, RequestDTO requestDTO);

	ResultDTO updateAircraft(AircraftDTO aircraftDTO, RequestDTO requestDTO);

    Page<Aircraft> getAllAircrafts(Pageable pageable);

    Page<Aircraft> getAllAircrafts(Specification<Aircraft> spec, Pageable pageable);

	ResponseEntity<AircraftPageDTO> getAircrafts(AircraftSearchDTO aircraftSearchDTO);
	
	List<AircraftDTO> convertAircraftsToAircraftDTOs(List<Aircraft> aircrafts, AircraftConvertCriteriaDTO convertCriteria);

	AircraftDTO getAircraftDTOById(Integer aircraftId);







}

File: skyspaceoneoneone_ui/src/services/AircraftService.js
import http from "../http-common"; 

class AircraftService {
  getAllAircrafts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/aircraft/aircrafts`, searchDTO);
  }

  get(aircraftId) {
    return this.getRequest(`/aircraft/${aircraftId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/aircraft?field=${matchData}`, null);
  }

  addAircraft(data) {
    return http.post("/aircraft/addAircraft", data);
  }

  update(data) {
  	return http.post("/aircraft/updateAircraft", data);
  }
  
  uploadImage(data,aircraftId) {
  	return http.postForm("/aircraft/uploadImage/"+aircraftId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AircraftService();
